director/redirect.go

```diff
@@ -440,14 +473,14 @@ func registerServeAd(engineCtx context.Context, ctx *gin.Context, sType ServerTy
 			}
 			if !ok {
 				log.Warningf("%s %v advertised to namespace %v without valid token scope\n",
-					sType, ad.Name, namespace.Path)
+					sType, adV2.Name, namespace.Path)
 				ctx.JSON(http.StatusForbidden, gin.H{"error": "Authorization token verification failed. Token missing required scope"})
 				return
 			}
 		}
 	} else {
 		token := strings.TrimPrefix(tokens[0], "Bearer ")
-		prefix := path.Join("caches", ad.Name)
+		prefix := path.Join("caches", adV2.Name)
```

Comment:
```suggestion
		prefix := path.Join("/caches", adV2.Name)
```

Reply:
Looking at git blame I think this causes error because of Justin's recent PR #583 that updated `GetNSIssuerURL` to check if the prefix comes with a slash. It returns error if it's not. So not directly related to your change